Add radio widget migration guide (#12080)
flutter/website: src/content/release/breaking-changes/radio-api-redesign.md

---
title: Redesigned the Radio Widget
description: >-
  Changes to the Radio widget
---

## Summary

Introduced the `RadioGroup` widget to centralize `groupValue` management and the `onChanged`
callback for a set of `Radio` widgets. As a result, the individual `Radio.groupValue` and
`Radio.onChanged` properties have been deprecated.

## Context

> To meet APG (ARIA Practices Guide) requirements for keyboard navigation and semantic properties in
> radio button groups, Flutter needed a dedicated radio group concept. Introducing a wrapper widget,
> RadioGroup, provides this out-of-the-box support. This change also presented an opportunity to
> simplify the API for individual Radio widgets.

Reviewer: ```suggestion
To meet APG (ARIA Practices Guide) requirements for keyboard navigation and
semantic properties in radio button groups, Flutter needed a dedicated radio group concept.
Introducing a wrapper widget, `RadioGroup`, provides this out-of-the-box support.
This change also presented an opportunity to simplify the API for individual `Radio` widgets.
```

## Description of change

> Deprecated `Radio.onChanged`, `Radio.groupValue`, `CupertinoRadio.onChanged`,
> `CupertinoRadio.groupValue`, `RadioListTile.groupValue`, and `RadioListTile.onChanged`.

Reviewer: ```suggestion
The following API is deprecated:
* `Radio.onChanged`
* `Radio.groupValue`
* `CupertinoRadio.onChanged`
* `CupertinoRadio.groupValue`
* `RadioListTile.groupValue`
* `RadioListTile.onChanged`.
```

## Migration guide

> If you are using these properties, you can refactor them with `RadioGroup`

Reviewer: ```suggestion
If you are using these properties, you can refactor them with `RadioGroup`.
```
